docs/tutorials/03_shaders/viz_sdf_cylinder.py

```diff
@@ -0,0 +1,296 @@
+
+vec_to_vec_rot_mat = import_fury_shader(
+    os.path.join('utils', 'vec_to_vec_rot_mat.glsl'))
+
+###############################################################################
+# We calculate the distance using the SDF function for the cylinder.
+
+sd_cylinder = import_fury_shader(os.path.join('sdf', 'sd_cylinder.frag'))
+
+###############################################################################
+# This is used on calculations for surface normals of the cylinder.
+
+sdf_map = \
+    '''
+    float map(in vec3 position)
+    {
+        // the sdCylinder function creates vertical cylinders by default, that
+        // is the cylinder is created pointing in the up direction (0, 1, 0).
+        // We want to rotate that vector to be aligned with the box's direction
+        mat4 rot = vec2VecRotMat(normalize(directionVSOutput),
+                                 normalize(vec3(0, 1, 0)));
+
+        vec3 pos = (rot * vec4(position - centerMCVSOutput, 0.0)).xyz;
+
+        // distance to the cylinder's boundary
+        return sdCylinder(pos, radiusVSOutput, heightVSOutput / 2);
+    }
+    '''
+
+###############################################################################
+# We use central differences technique for computing surface normals.
+
+central_diffs_normal = import_fury_shader(os.path.join('sdf',
+                                                       'central_diffs.frag'))
+
+###############################################################################
+# We use cast_ray for the implementation of Ray Marching.
+
+cast_ray = import_fury_shader(os.path.join('ray_marching', 'cast_ray.frag'))
+
+###############################################################################
+# For the illumination of the scene we use the Blinn-Phong model.
+
+blinn_phong_model = import_fury_shader(os.path.join('lighting',
+                                                    'blinn_phong_model.frag'))
+
+###############################################################################
+# Now we use compose_shader to join our pieces of GLSL shader code.
+
+fs_dec = compose_shader([fs_vars_dec, vec_to_vec_rot_mat, sd_cylinder, sdf_map,
+                         central_diffs_normal, cast_ray, blinn_phong_model])
+
+shader_to_actor(box_actor, "fragment", decl_code=fs_dec)
+
+###############################################################################
+# Here we have the implementation of all the previous code with all the
+# necessary variables and functions to build the cylinders.
+
+sdf_cylinder_frag_impl = \
+    '''
+    vec3 point = vertexMCVSOutput.xyz;
+
+    // ray origin
+    vec4 ro = -MCVCMatrix[3] * MCVCMatrix;  // camera position in world space
+
+    // ray direction
+    vec3 rd = normalize(point - ro.xyz);
+
+    // light direction
+    vec3 ld = normalize(ro.xyz - point);
+
+    ro += vec4((point - ro.xyz), 0);
+
+    float t = castRay(ro.xyz, rd);
+
+    if(t < 20.0)
+    {
+        vec3 position = ro.xyz + t * rd;
```

Comment:
@tvcastillod Since the intersection point is already there you might as well use it to calculate the zdepth to allow intersection with each actor and with other actors as well. Please see the example below:
https://github.com/fury-gl/fury/blob/294d01ce6d3afd70f4c967f3336a84391cdb5bfb/fury/shaders/gs_billboard_sphere_impl.frag#L32-L33

Reply:
@m-agour thanks for your suggestion. As this is a tutorial I don't want to go into too much detail so as not to confuse the user, but it would be nice to keep it in mind for a future tutorial.